llvm/include/llvm/CodeGen/SelectionDAG.h

```diff
@@ -247,6 +248,9 @@ class SelectionDAG {
   BlockFrequencyInfo *BFI = nullptr;
   MachineModuleInfo *MMI = nullptr;
 
+  /// Extended EVTs used for single value VTLists.
+  std::set<EVT, EVT::compareRawBits> EVTs;
```

Comment:
Did you try all the set types? Does DenseSet do any better?

Reply:
I didn't try DenseSEt. Looks like we would need to add a DenseMapInfo for EVT including defining Empty and Tombstone values?

Reply:
Yes

Reply:
The usage here depends on the address stability of the map entries (the VTList points at the map's key), and DenseSet doesn't provide that.  So you'd need define a struct like `EVTSetKey Key { EVT* Ptr; };`, define DenseMapInfo for that, then use `DenseSet<EVTSetKey>`.

I guess you could do that, but it seems like overkill.